index.rst

```diff
@@ -78,11 +76,31 @@ This is the root object for XGBoost model.
 .. parsed-literal::
 
   {
-    "major_version" : 1,
-    "minor_version" : 0,
+    "version" : [1, 0],
+    "node_schema" : [
+      ["parent", "integer"],
```

Comment:
Is it possible to save it as `["parent", "cleft", "cright", "sindex", "default_left", "float_info"]`? Since I think the type info can be easily deduced.

Reply:
Should they be deduced? Are they actually needed? What would happen if we pass a `bool` into a model trained with `float` for this value? I don't remember now why I have started saving theit types.

Reply:
>since the XGBoost codebase will enforce the field names and types already.

Do types of columns have any impact on the resulting file? The types are not saved in the file, but do they affect training and inference? 

I mean, let's assume that ensemble building is completely deterministic. Then, what will be different in the resulting file if we feed the learner with a dataset of vectors of 1 and 0, or the same dataset, but 1. and 0. and true and false as values? And what gonna happen if we train on a dataset of floats but then load the model, setting types into int or bool, convert the dataset into them and infer on them?

Order really matters, but do types?

Reply:
I should clarify. The schemas defined here are for saving model information, such as nodes. The types only apply to the serialized models, not the input data. Currently all inputs are assumed to be float. (Python wrapper performs necessary conversion)

Reply:
We didn’t decide yet whether feature names and types should be saved. Right now my proposal doesn’t, since the current binary serialization does not.